### PR TITLE
Feature/5586 event check in processing key submission

### PR DESCRIPTION
**Pull request by whiskey**
## Description
Adds (valid) checkins to the key submission payload.

## Link to Jira
[EXPOSUREAPP-5688](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-5688)
others tbd

## Screenshots
n/a

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Extensions/ENTemporaryExposureKey+Processing.swift [hunk 1]
```diff
@@ -5,6 +5,7 @@
 import Foundation
 import ExposureNotification
 
+// [KGA] Hier wäre Ziel, processedforSubmission zu nutzen für meine Keys. Wie alt ist der Checkin. Dann entsprechend transmissionRiskLevel infektion zu weisen
```
Comment (naveeddotio) [inline src/xcode/ENA/ENA/Source/Extensions/ENTemporaryExposureKey+Processing.swift:8]: Can we change the comment to English?

Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Extensions/ENTemporaryExposureKey+Processing.swift:8]: Oh we missed that one. All `[KGA]` comments will be removed :)

File: src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift [hunk 2]
```diff
@@ -154,12 +155,21 @@ class ENAExposureSubmissionService: ExposureSubmissionService {
...
+		appConfigurationProvider
+			.appConfiguration()
+			.sink { appConfig in
+				// Fetch & process keys and checkins
+				let processedKeys = keys.processedForSubmission(with: self.symptomsOnset)
+				self.preparedCheckinsForSubmission(with: appConfig, symptomOnset: self.symptomsOnset, completion: { checkins in
+					// Request needs to be prepended by the fake request.
+					self._fakeVerificationServerRequest(completion: { _ in
+						self._submitExposure(processedKeys, visitedCountries: self.supportedCountries, attendedEvents: checkins, completion: completion)
+					})
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift:169]: This means that we have for the checkins submission fake requests, too? Do we need that?

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+SubmitTests.swift [hunk 3]
```diff
@@ -172,7 +172,7 @@ final class HTTPClientSubmitTests: XCTestCase {
 		let expectation = self.expectation(description: "Response403")
 
 		// Act
-		let payload = CountrySubmissionPayload(exposureKeys: keys, visitedCountries: [], tan: tan)
+		let payload = CountrySubmissionPayload(exposureKeys: keys, visitedCountries: [], eventCheckIns: [], tan: tan)
 		HTTPClient.makeWith(mock: stack).submit(payload: payload, isFake: false) { response in
 			defer { expectation.fulfill() }
```
Comment (pascal-brause) [inline src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+SubmitTests.swift:178]: Are there missing tests for not empty lists of eventCheckIns?

File: src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService+CheckinSubmission.swift [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+	///
+	///  Details on the implementation can be found in the [tech spec](https://github.com/corona-warn-app/cwa-app-tech-spec/blob/proposal/event-registration-mvp/docs/spec/event-registration-client.md#attendee-check-in-submission).
+	/// - Returns: A list of converted checkins
+	func preparedCheckinsForSubmission(with appConfig: SAP_Internal_V2_ApplicationConfigurationIOS, symptomOnset: SymptomsOnset, completion: @escaping (_ transformed: [SAP_Internal_Pt_CheckIn]) -> Void) {
+		let eventStore = EventStore(url: EventStore.storeURL)
+		let raw🐓 = eventStore?.checkinsPublisher.value ?? []
+		let css = CheckinSplittingService()
+		let transmissionRiskValueMapping = appConfig.presenceTracingParameters.riskCalculationParameters.transmissionRiskValueMapping
+
+		let checkins = raw🐓
```
[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService+CheckinSubmission.swift:19]: raw-chicken sounds like "raw-checkin" or vice versa ;-)

[the remark this answers is not in the document]
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService+CheckinSubmission.swift]: Renamed 😉

File: src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift (hunk 2 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Services/Exposure Submission/ExposureSubmissionService.swift:169]: We just have enhanced these lines with our check-in stuff. So the surrounding fake requests already have been there. So we didn't changed that behavior.

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+SubmitTests.swift (hunk 3 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+SubmitTests.swift:178]: The tests are work in progress. My plan to add them today was delayed but I'm on them.

**Review by SwissMountainFrog — APPROVED** [review]

File: src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+SubmitTests.swift (hunk 3 above)
Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Client/HTTP Client/__tests__/HTTPClient+SubmitTests.swift:178]: I checked the existing tests and found out a couple of issues with them:
* All tests use the `MockNetworkStack` which defines the response. Providing multiple CheckIns wouldn't change anything as there are no sanity checks for the provided data
* None of the tests provides `visitedCountries` as well. I guess for the aforementioned reason.

Generally we should consider live-server tests with 'real' payloads to test our code under real conditions. Happy to discuss this after the vacation. For now I would keep this in mind (and noted) but wouldn't extend these tests more than necessary.

**Review by naveeddotio — APPROVED** [review]
